domain/model/service/modelservice.go

```diff
@@ -94,5 +107,58 @@ func (s *ModelService) CreateModel(
 func (s *ModelService) DeleteModel(
 	ctx context.Context,
 ) error {
-	return s.st.Delete(ctx, s.modelID)
+	return s.modelSt.Delete(ctx, s.modelID)
+}
+
+// Status returns the status of the model.
+func (s *ModelService) Status(ctx context.Context) (model.StatusInfo, error) {
+	modelState, err := s.controllerSt.GetModelState(ctx, s.modelID)
+	if err != nil {
+		return model.StatusInfo{}, err
+	}
+	if modelState.Destroying {
+		return model.StatusInfo{
+			Status:  corestatus.Destroying,
+			Message: "the model is being destroyed",
+		}, nil
+	}
+	if modelState.InvalidCloudCredentialReason != "" {
```

Comment:
I don't like writing boolean logic based on if a string is empty or not. The database has a boolean for this exact case so this struct should have a boolean to perform branching logic off of.

Would you consider moving this if statement to the top of the func? Based on my read of the code this would be the first check that is currently performed in Mongo and so I think it best to maintain this type of logic for the moment till we can understand subtle breakages.